internal/dkg/actions_signing.go

```diff
@@ -33,14 +36,17 @@ func (d *Process) signMessage(
 	}, nil
 }
 
-func (d *Process) verifyMessage(packet *drand.GossipPacket, metadata *drand.GossipMetadata, proposal *drand.ProposalTerms) error {
-	participants := util.Concat(proposal.Remaining, proposal.Joining)
+func (d *Process) verifyMessage(packet *drand.GossipPacket, proposal *drand.ProposalTerms) error {
+	beaconID := packet.GetMetadata().GetBeaconID()
```

Comment:
not sure if it's possible here, but might be worth checking `packet.GetMetadata != nil`

Reply:
this is done in the protobuf method inside of  `GetBeaconID()`